### PR TITLE
Add 2FA recovery code display on enrollment

### PR DESCRIPTION
**Pull request by mc-jones**
Feature to automatically show 2FA backup recovery codes after successful enrollment. Discussed in PR https://github.com/artsy/gravity/pull/13099. 

**TO-DOs:** 
- Add copy/download functionality on display modal.

**Considerations for future:**
- When I removed all 2FA options and reenrolled, I received the same recovery codes as the prior enrollment. I would expect that if I removed all 2FA accounts that the previous recovery codes would be deleted automatically. @joeyAghion I saw your and @dblandin's discussion on the linked PR but I'm thinking of slightly different use case. Specifically if a user enrolled in 2FA, then at some point disables it and never reenroll's. I think it's worth considering the value of  holding onto potentially sensitive data for a feature the User has chosen not to use.

Jira: https://artsyproduct.atlassian.net/browse/PLATFORM-2412?atlOrigin=eyJpIjoiMmUyYjBkNWNmODVmNGM1MDgwNGRmYmJlM2QxOGRjN2QiLCJwIjoiaiJ9

### CODE REVIEW

File: src/v2/Components/UserSettings/TwoFactorAuthentication/Components/SmsSecondFactor/Modal.tsx [hunk 1]
```diff
@@ -46,12 +50,16 @@ export const SmsSecondFactorModal: React.FC<SmsSecondFactorModalProps> = props =
...
 
     try {
-      await EnableSecondFactor(relayEnvironment, {
+      const response = await EnableSecondFactor(relayEnvironment, {
         secondFactorID: secondFactor.internalID,
         code: values.code,
       })
 
-      onComplete()
+      console.log("RESPONSE", response)
```
Comment (dzucconi) [inline src/v2/Components/UserSettings/TwoFactorAuthentication/Components/SmsSecondFactor/Modal.tsx:58]: 👋

File: src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactorReminder.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  backupSecondFactors: string[]
+  factorTypeName: string
+}
+
+export const BackupSecondFactorReminder: React.FC<BackupSecondFactorReminderProps> = props => {
+  const { backupSecondFactors, factorTypeName } = props
+
+  return (
+    <Box minHeight="280px">
+      <Sans size="3" color="black60">
```
Comment (dzucconi) [inline src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactorReminder.tsx:14]: The `Sans` component is deprecated and shouldn't be used. You can [use `Text` and V2 variants here](https://palette-storybook.artsy.net/?path=/story/components-text--variants)

**Comment by dzucconi** [comment]
Can you add a screenshot of what this looks like?

**Review by joeyAghion — APPROVED** [review]
Re. the future consideration, it sounds like you're suggesting disabling any recovery codes as part of the process of disabling the account's "last" 2nd factor (in [this mutation](https://github.com/artsy/gravity/blob/master/app/graphql/mutations/disable_second_factor.rb)).

That doesn't sound unreasonable to me. And the re-enrolling process would hopefully just work (i.e., [generate new recovery codes](https://github.com/artsy/gravity/blob/748bb130b6a41b12534eecb8878664239f8218e0/app/graphql/mutations/enable_second_factor.rb#L22) if none exist).

File: src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactorReminder.tsx (hunk 2 above)
Comment (mc-jones) [reply src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactorReminder.tsx]: Noted. I've changed them in this Components and have added a chore task on the Epic to switch on the other 2FA Components to use `Text`.

**Comment by mc-jones** [comment]
@dzucconi - screenshot: https://drive.google.com/file/d/1CIqv289hnRC_9dst6Ziio-33FM9ZTbeH/view?usp=sharing

**Review by dzucconi — APPROVED** [review]

**Comment by damassi** [comment]
@mc-jones - Thanks for adding the link! but if possible, we generally add screenshots embedded directly in to the description so that folks passing by in PR review can quickly see the change.

**Comment by mc-jones** [comment]
@damassi - no problem, makes sense! Definitely will do. I was just being overly cautious since this is a public repo and it included my active recovery keys for the test account I was using

**Pull request merged**